solr/solrj/src/java/org/apache/solr/client/solrj/impl/CloudHttp2SolrClient.java

```diff
@@ -68,7 +69,21 @@ protected CloudHttp2SolrClient(Builder builder) {
             "Both zkHost(s) & solrUrl(s) have been specified. Only specify one.");
       }
       if (builder.zkHosts != null) {
-        this.stateProvider = new ZkClientClusterStateProvider(builder.zkHosts, builder.zkChroot);
+        try {
```

Comment:
Worth some docs on this?   Why do we need to use the Class.forName, is this potentially pluggable somehow?  Makes me think of JDBC drivers!

Reply:
I'll add a line -- it's rather fundamental to this PR.

For user level, I added a note to solr-upgrade-notes.adoc